src/renderer/utils/InstallManager.tsx

```diff
@@ -688,11 +600,192 @@ export class InstallManager {
     this.setCurrentlyInstalledTrack(addon, null);
   }
 
+  private static getAddonInstall(directory: string): InstallManifest | null {
+    try {
+      return getCurrentInstall(directory);
+    } catch (e) {
+      return null;
+    }
+  }
+
+  public static async getAddonInstallState(addon: Addon): Promise<InstallState> {
+    const status = store.getState().installStatus[addon.key] as InstallState;
+
+    if (status) {
+      return status;
+    }
+
+    return this.refreshAddonInstallState(addon);
+  }
+
+  public static async refreshAddonInstallState(addon: Addon): Promise<InstallState> {
+    const currenState = store.getState().installStatus[addon.key] as InstallState;
```

Comment:
```suggestion
    const currentState = store.getState().installStatus[addon.key] as InstallState;
```
typo :D (don't commit this suggestion)

Reply:
Fixed.